tests/core/CoreTest.py

```diff
@@ -383,7 +383,8 @@ def execute_run(bears):
         def on_result(result):
             results.append(result)
 
-        run(bears, on_result)
+        session = Session(bears, on_result)
+        session.run()
```

Comment:
If you want to do this (I think we should do this :D), then mention this inside the commit message :+1: